sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -57,9 +57,29 @@ object Row {
   /**
    * Merge multiple rows into a single row, one after another.
    */
+
+  @deprecated("This method is deprecated and will be removed in future versions.", "3.0.0")
   def merge(rows: Row*): Row = {
-    // TODO: Improve the performance of this if used in performance critical part.
-    new GenericRow(rows.flatMap(_.toSeq).toArray)
+    var fSize, sSize = rows.size
+    var number = 0
+
+    while (fSize != 0) {
+      fSize -= 1
+      number += rows(fSize).size
+    }
+
+    val container = Array.ofDim[Any](number)
+
+    while (sSize != 0) {
+      sSize -= 1
+      var ssSize = rows(sSize).size
+      while (ssSize != 0) {
+        number -= 1
+        ssSize -= 1
+        container(number) = rows(sSize)(ssSize)
+      }
+    }
```

Comment:
I would just leave the function as was. It will be removed in the next release anyway.

Reply:
yea, it is not used before and is deprecated now. We don't need to do this optimization.